beacon-chain/slasher/detect_attestations.go

```diff
@@ -0,0 +1,474 @@
+package slasher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	slashertypes "github.com/prysmaticlabs/prysm/beacon-chain/slasher/types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/time/slots"
+	"go.opencensus.io/trace"
+)
+
+// Takes in a list of indexed attestation wrappers and returns any
+// found attester slashings to the caller.
+func (s *Service) checkSlashableAttestations(
+	ctx context.Context, atts []*slashertypes.IndexedAttestationWrapper,
+) ([]*ethpb.AttesterSlashing, error) {
+	currentEpoch := slots.EpochsSinceGenesis(s.genesisTime)
+	slashings := make([]*ethpb.AttesterSlashing, 0)
+	indices := make([]types.ValidatorIndex, 0)
+
+	// TODO(#8331): Consider using goroutines and wait groups here.
+	groupedAtts := s.groupByValidatorChunkIndex(atts)
+	for validatorChunkIdx, batch := range groupedAtts {
+		attSlashings, err := s.detectAllAttesterSlashings(ctx, &chunkUpdateArgs{
+			validatorChunkIndex: validatorChunkIdx,
+			currentEpoch:        currentEpoch,
+		}, batch)
+
+		slashings = append(slashings, attSlashings...)
+		if err != nil {
```

Comment:
this err is at wrong place

Reply:
Seems good to me. Whats wrong with it? @terencechain

Reply:
the err is after the `append`, it should be before the `append`

Reply:
It's fixed in https://github.com/prysmaticlabs/prysm/pull/9694/commits/6eeb088a69d4307c25966610e311ba9d823ac84b